web-ui/src/main/resources/catalog/js/CatController.js

```diff
@@ -679,7 +679,17 @@ goog.require('gn_alert');
               });
         });
 
-
+        // Retreive portal information
+        var getPortalSources = promiseStart.then(function(value) {
```

Comment:
The site API should take care of this https://github.com/geonetwork/core-geonetwork/blob/4.0.x/services/src/main/java/org/fao/geonet/api/site/SiteApi.java#L192-L207

Node id and name should be set properly https://apps.titellus.net/geonetwork/wise/api/site

Also some setup have more than 200 portals and this approach would be slow.

Reply:
Thanks, will try to rewrite it a bit.

(200 portals?)

Reply:
It is not really about rewriting something. It should work eg. https://apps.titellus.net/geonetwork/wise/eng/catalog.search#/home maybe something which is not in master or 3 ?

Reply:
I mean rewriting the toolbar file.

Reply:
The code in https://github.com/geonetwork/core-geonetwork/blob/3028d14b9d16b146a8e46c5ff1217e865d49c9cf/services/src/main/java/org/fao/geonet/api/site/SiteApi.java#L209-L224 seem the same

The value `"node/name"` of the response for `/api/site` seem should have the translated node value.

Reply:
I had another look at some code, and saw this line: https://github.com/geonetwork/core-geonetwork/blob/master/services/src/main/java/org/fao/geonet/api/site/SiteApi.java#L209

It looks like (I'm no Java developer) the `NODE` information is only included when you're not on the 'default' node/site. It would have been nice if this information was always there, but maybe there is a good reason not to do so.

Reply:
> if this information was always there,

Yep could be easier.